src/include/function_table_initializer.hpp

```diff
@@ -59,14 +59,20 @@ class table_initializer {
     using dlhandle = std::unique_ptr<LIB_TYPE, handle_deleter>;
 
 public:
-    function_table_t &operator[](oneapi::mkl::device key) {
-        auto lib = tables.find(key);
+    function_table_t &operator[](std::pair<oneapi::mkl::device, sycl::queue> device_queue_pair) {
```

Comment:
It's not clear to me why this change is necessary. Where do we need the `sycl::queue` in the lookup tables?

Reply:
Since I moved the `unsupported_device` exception in the table initializer, I need the queue to query the device name. This is the only reason why I need the `sycl::queue` here.

Reply:
That makes sense. It seems like a large change for just the exception - I would support reverting the function table and throwing a more generic exception (or maybe implement `mkl::unsupported_device` with no device argument?). But it is also okay as is.